pkg/controllers/nodeclass/validation.go

```diff
@@ -331,12 +331,20 @@ func (v *Validation) mockOptions(ctx context.Context, nodeClaim *karpv1.NodeClai
 	}
 	return v.amiResolver.Resolve(nodeClass, nodeClaim, []*cloudprovider.InstanceType{
 		{
-			Name:         "m5.large",
-			Requirements: scheduling.NewRequirements(scheduling.NewRequirement(corev1.LabelArchStable, corev1.NodeSelectorOpIn, karpv1.ArchitectureAmd64)),
+			Name: "m5.large",
+			Requirements: scheduling.NewRequirements(
```

Comment:
I don't love that we need to hardcode these requirements here, but I don't think there's a better option since we're hardcoding the instance types used for RunInstances validation. We could fix this by dynamically resolving the instance types from the EC2NodeClass, but I don't think we need to scope that into this PR - I'm actually working on a PR that does that already.